Bet Tracker View

JiwonJeong414 opened the pull request:
Implemented the Bet Tracker View with custom charts and cards. Mostly used Peter's code for dummy data which was really helpful. The helper functions for calculating the gains and losses might be incorrect (need review) 

Todo:
- add custom back button for the FAQ
- better charts styling for monthly view
- add the final cards in the button after those are done
- replace dummy data with actual data

Ant's Dummy Data
<img width="429" alt="Screenshot 2025-03-20 at 7 55 51 PM" src="https://github.com/user-attachments/assets/bc3abfeb-d961-4b29-bc27-a170b6d12f44" />

Peter's Dummy Data
<img width="429" alt="Screenshot 2025-03-20 at 7 56 48 PM" src="https://github.com/user-attachments/assets/64ae2b05-497b-4f9e-8fff-ab3a6392dece" />

Video:

https://github.com/user-attachments/assets/28c34797-6989-4bc5-89a4-fa080a8ca27c

MrPeterss committed:
refactor tab bar

JiwonJeong414 committed:
General Layout

JiwonJeong414 committed:
Major Changes to Styling

antoinztte commented on All In/Features/Profile/BetTrackerView.swift:
```diff
@@ -0,0 +1,348 @@
+            VStack(spacing: 16) {
+                trackerCard
+
+                HStack(spacing: 17) {
```
Why are we using 17 as spacing for these?

JiwonJeong414 replied:
on the figma file, the spacing is 17

<img width="584" alt="image" src="https://github.com/user-attachments/assets/cd553447-e92d-4e6a-97ef-2cc84dff0dbf" />

antoinztte commented on All In/Features/Shared/Models/UserModel.swift:
```diff
@@ -5,41 +5,84 @@
         ),
         User(
             id: 1,
-            username: "antoinztte",
```
Wowwwwwwww

antoinztte reviewed (dismissed):
Let me speak w/ designers about the Monthly view of the chart + the specific sizing before you push this. Everything else LGTM!

antoinztte commented:
Let's fix the back arrow for FAQ page to reflect the Figma!

MrPeterss committed:
Merge branch 'jiwon/bet-tracker-page' into peter/tab-view-refactor

MrPeterss committed:
Update ContentView.swift

MrPeterss committed:
Update TabBar.swift

MrPeterss committed:
update tabbar

MrPeterss committed:
Update .gitignore

JiwonJeong414 committed:
Merge pull request #15 from cuappdev/peter/tab-view-refactor

tab view refactor

JiwonJeong414 committed:
Merge branch 'main' into jiwon/bet-tracker-page

JiwonJeong414 committed:
Addressed Ant's Comments

JiwonJeong414 committed:
Minor Fix

JiwonJeong414 committed:
Merge branch 'main' into jiwon/bet-tracker-page

JiwonJeong414 committed:
Revert "Merge branch 'main' into jiwon/bet-tracker-page"

This reverts commit ecc5e7597af832bcfe017580b9c1a1e2315a4e83, reversing
changes made to 952fcadab717e0f599c42746fc4f37f4c4c549f5.

JiwonJeong414 commented:
Updated View:

- fixed monthly view
- fixed faq view
- fixed back button view


https://github.com/user-attachments/assets/b2de9e97-0a1c-4f22-b481-99295144b54f

MrPeterss commented on All In/AllInApp.swift:
```diff
@@ -9,10 +9,12 @@ import SwiftUI
+                .environmentObject(TabNavigationManager())
         }
     }
 }
```
lets add a newline above this line as well

MrPeterss commented on All In/Features/Home/HomeView.swift:
```diff
@@ -79,44 +79,45 @@ struct HomeView: View {
-                        .resizable()
-                        .foregroundStyle(Constants.Colors.grey03)
-                        .frame(width: 42, height: 42)
+        (
```
what is the reason for the parenthesis? lets remove these

MrPeterss commented on All In/Features/Home/HomeView.swift:
```diff
@@ -79,44 +79,45 @@ struct HomeView: View {
-                        .frame(width: 42, height: 42)
+        (
+            Button {
+
```
lets make this button go to the market page. add an environment reference to the "tabNavigationManager" at the top of the struct and then in the button action lets do:

"tabNavigationManager.selectedTab = .market"

MrPeterss commented on All In/Features/Home/HomeView.swift:
```diff
@@ -79,44 +79,45 @@ struct HomeView: View {
+                        .background(.white)
+                        .cornerRadius(16)
+                        .shadow(color: Constants.Colors.grey00, radius: 5, x: 0, y: 4)
+                    VStack {
```
can we add a newline above?

MrPeterss commented on All In/Features/Home/Views/FrequentAskedQuestion.swift:
```diff
@@ -44,11 +56,11 @@ extension FrequentAskedQuestion {
         VStack(alignment: .leading, spacing: 8) {
             Text("How can we help?")
-                .font(Constants.Fonts.faq)
+                .font(Constants.Fonts.headerProfile)
```
we should think about a more standardized naming convention for our constants. we can do that another time

MrPeterss commented on All In/Features/Profile/BetTrackerView.swift:
```diff
@@ -0,0 +1,339 @@
+
+    let user: User
+
+    var myGradient = Gradient(
```
lets put this in the constants file... seems like we have to use it elsewhere

MrPeterss reviewed (approved):
Really nice work Jiwon! I trust you can fix the few comments I left before merging. Keep it up!

JiwonJeong414 committed:
Addressed Peter's Comments

AdelynnWu reviewed (approved):
looks good to me!
